apps/web/app/(app)/environments/[environmentId]/surveys/components/SurveyCard.tsx

```diff
@@ -0,0 +1,147 @@
+import { generateSingleUseIdAction } from "@/app/(app)/environments/[environmentId]/surveys/actions";
+import { TSurveyListItem } from "@/app/(app)/environments/[environmentId]/surveys/types/survey";
+import { Code, EarthIcon, Link2Icon } from "lucide-react";
+import Link from "next/link";
+import { useEffect, useMemo, useState } from "react";
+import toast from "react-hot-toast";
+import { getFormattedErrorMessage } from "@formbricks/lib/actionClient/helper";
+import { cn } from "@formbricks/lib/cn";
+import { convertDateString, timeSince } from "@formbricks/lib/time";
+import { TEnvironment } from "@formbricks/types/environment";
+import { SurveyStatusIndicator } from "@formbricks/ui/components/SurveyStatusIndicator";
+import { SurveyDropDownMenu } from "./SurveyDropdownMenu";
+
+interface SurveyCardProps {
+  survey: TSurveyListItem;
+  environment: TEnvironment;
+  otherEnvironment: TEnvironment;
+  isViewer: boolean;
+  WEBAPP_URL: string;
+  duplicateSurvey: (survey: TSurveyListItem) => void;
+  deleteSurvey: (surveyId: string) => void;
+}
+export const SurveyCard = ({
+  survey,
+  environment,
+  otherEnvironment,
+  isViewer,
+  WEBAPP_URL,
+  deleteSurvey,
+  duplicateSurvey,
+}: SurveyCardProps) => {
+  const isSurveyCreationDeletionDisabled = isViewer;
+
+  const surveyStatusLabel = useMemo(() => {
+    if (survey.status === "inProgress") return "In Progress";
+    else if (survey.status === "scheduled") return "Scheduled";
+    else if (survey.status === "completed") return "Completed";
+    else if (survey.status === "draft") return "Draft";
+    else if (survey.status === "paused") return "Paused";
+  }, [survey]);
+
+  const [singleUseId, setSingleUseId] = useState<string | undefined>();
+
+  useEffect(() => {
```

Comment:
can we also generate this singleUseId in the dropdown itself and make this whole component a server component?

Reply:
`SurveyCard` is called by `SurveyList` which is a client component so, this would automatically make this a client component 😊